src/main/groovy/liquibase/harness/base/BaseCompatibilityTest.groovy

```diff
@@ -0,0 +1,225 @@
+package liquibase.harness.base
+
+import liquibase.Scope
+import liquibase.database.jvm.JdbcConnection
+import liquibase.harness.config.DatabaseUnderTest
+import liquibase.harness.config.TestConfig
+import liquibase.harness.util.rollback.RollbackStrategy
+import org.json.JSONArray
+import org.json.JSONObject
+import org.junit.Assert
+import org.skyscreamer.jsonassert.JSONCompareMode
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import java.sql.Connection
+import java.sql.DriverManager
+import java.sql.ResultSet
+import java.sql.SQLException
+
+import static liquibase.harness.util.FileUtils.*
+import static liquibase.harness.util.JSONUtils.*
+import static liquibase.harness.util.TestUtils.*
+import static liquibase.harness.base.BaseCompatibilityTestHelper.*
+
+@Unroll
+class BaseCompatibilityTest extends Specification {
+    @Shared
+    RollbackStrategy strategy;
+    @Shared
+    List<DatabaseUnderTest> databases;
+
+    def setupSpec() {
+        databases = TestConfig.instance.getFilteredDatabasesUnderTest()
+        strategy = chooseRollbackStrategy()
+        strategy.prepareForRollback(databases)
+    }
+
+    def "apply #testInput.change against #testInput.databaseName #testInput.version"() {
+        given: "read input data"
+        String expectedResultSet = getJSONFileContent(testInput.change, testInput.databaseName, testInput.version,
+                "liquibase/harness/base/expectedResultSet")
+        Map<String, Object> argsMap = new HashMap()
+        argsMap.put("url", testInput.url)
+        argsMap.put("username", testInput.username)
+        argsMap.put("password", testInput.password)
+        String basePath = "liquibase/harness/base/"
+        String pathToXmlChangelogFile = "${basePath}changelogs/${testInput.change}.xml"
```

Comment:
We can create Array or Collection for these 4 values and iterate over them in loop 4 times for all operations

Reply:
++